File: backend/src/internal/models/bank.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Bank struct {
	ID uint `json:"id"`

	Name     string `gorm:"unique;not null" json:"name"`
	OrderNum uint   `json:"orderNum" gorm:"default:1"`

	Status bool `json:"status"`

	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

File: frontend/src/src/actions/bank.ts
import { gql } from "@apollo/client";

export const FILTER_BANK = gql`
  query GetBanks(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: getBanks(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      banks {
        id
        name
        orderNum
        status
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const CREATE_BANK = gql`
  mutation NewBank($input: NewBankInput!) {
    response: createBank(input: $input) {
      id
      name
      orderNum
      status
      createdAt
      updatedAt
      deletedAt
    }
  }
`;

export const UPDATE_BANK = gql`
  mutation UpdateBank($id: ID!, $input: UpdateBankInput!) {
    response: updateBank(id: $id, input: $input) {
      id
      name
      orderNum
      status
      createdAt
      updatedAt
      deletedAt
    }
  }
`;

export const DELETE_BANK = gql`
  mutation DeleteBank($id: ID!) {
    response: deleteBank(id: $id)
  }
`;
